ui/components/app/gas-timing/gas-timing.component.js

```diff
@@ -8,28 +8,65 @@ import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
 import { I18nContext } from '../../../contexts/i18n';
 
 import Typography from '../../ui/typography/typography';
-import { TYPOGRAPHY } from '../../../helpers/constants/design-system';
+import {
+  TYPOGRAPHY,
+  FONT_WEIGHT,
+} from '../../../helpers/constants/design-system';
+import InfoTooltip from '../../ui/info-tooltip/info-tooltip';
+
+import { getGasTimeEstimate } from '../../../store/actions';
 
 // Once we reach this second threshold, we switch to minutes as a unit
 const SECOND_CUTOFF = 90;
 
-export default function GasTiming({ maxPriorityFeePerGas }) {
+// Shows "seconds" as unit of time if under SECOND_CUTOFF, otherwise "minutes"
+const toHumanReadableTime = (milliseconds = 1, t) => {
+  const seconds = Math.ceil(milliseconds / 1000);
+  if (seconds <= SECOND_CUTOFF) {
+    return t('gasTimingSeconds', [seconds]);
+  }
+  return t('gasTimingMinutes', [Math.ceil(seconds / 60)]);
+};
+export default function GasTiming({
+  maxFeePerGas = 0,
+  maxPriorityFeePerGas = 0,
+}) {
   const {
     gasFeeEstimates,
     isGasEstimatesLoading,
     gasEstimateType,
   } = useGasFeeEstimates();
 
-  const t = useContext(I18nContext);
+  const [customEstimatedTime, setCustomEstimatedTime] = useState(null);
 
-  // Shows "seconds" as unit of time if under SECOND_CUTOFF, otherwise "minutes"
-  const toHumanReadableTime = (milliseconds = 1) => {
-    const seconds = Math.ceil(milliseconds / 1000);
-    if (seconds <= SECOND_CUTOFF) {
-      return t('gasTimingSeconds', [seconds]);
+  // If the user has chosen a value lower than the low gas fee estimate,
+  // We'll need to use the useEffect hook below to make a call to calculate
+  // the time to show
+  const isUnknownLow =
+    gasFeeEstimates?.low &&
+    Number(maxPriorityFeePerGas) <
+      Number(gasFeeEstimates.low.suggestedMaxPriorityFeePerGas);
+
+  useEffect(() => {
+    const priority = maxPriorityFeePerGas;
+    const fee = maxFeePerGas;
+
+    if (isUnknownLow) {
```

Comment:
We probably also need to call `getGasTimeEstimate` in situations where `isUnkownLow` is false. For instance, if `maxPriorityFeePerGas` is greater than `gasFeeEstimates.low.suggestedMaxPriorityFeePerGas`, and then either `maxPriorityFeePerGas` or `maxFeePerGas` change